084-open-feature-integration.md

```diff
@@ -0,0 +1,417 @@
+                // feature gates config is null or empty so we are gonna retrieve it from ENV VAR
+                featureGatesConfig = this.featureClient.getStringValue(STRIMZI_FEATURE_GATES_ENV, "");
+            }
+
+            // parse the featureGateConfig if it's provided
+            this.validateFeatureGateConfig(featureGatesConfig);
+        } else {
+            // other providers (e.g., flagd)
+            this.continueOnManualRUFailure = evaluationContext != null ?
+                new FeatureGate(CONTINUE_ON_MANUAL_RU_FAILURE, fetchFeatureFlag(CONTINUE_ON_MANUAL_RU_FAILURE, false, Boolean.class, evaluationContext)) :
+                new FeatureGate(CONTINUE_ON_MANUAL_RU_FAILURE, fetchFeatureFlag(CONTINUE_ON_MANUAL_RU_FAILURE, false, Boolean.class));
+        }
+
+        // Validate interdependencies (if any)
+        this.validateInterDependencies();
+    }
+}
+```
+In the context of `EnvVarProvider`, it should be fairly simple to implement. 
+By using the new provider and adapting the current implementation, it should function the same as our existing approach.
+Alternatively, when using other `Provider` (e.g., FlagD), there are a couple of options: (i) an user can deploy just standalone application FlagD as a deployment, 
+or (ii) deploy the [OpenFeature Operator](https://github.com/open-feature/open-feature-operator), which supports FlagD as one of its flagging systems. 
+While researching, I also discovered several other feature flagging systems, such as:
+1. [Go Feature Flag](https://gofeatureflag.org/)
+2. [CloudBees Feature Management](https://www.cloudbees.com/capabilities/feature-management)
+3. [Split](https://www.split.io/)
+4. [Harness](https://harness.io/products/feature-flags)
+5. [LaunchDarkly](https://launchdarkly.com/)
+6. [Flagsmith](https://flagsmith.com/)
+7. [Flipt](https://www.flipt.io/)
+
+Users have the flexibility to choose any feature flagging provider that suits their needs, thanks to the integration with OpenFeature, which supports multiple providers.
+Given its community support and inclusion in the CNCF ecosystem, OpenFeature offers a versatile solution for feature management.
+
+Conceptually, the communication between Strimzi and a feature flagging system can be illustrated as:
+
+
+    +------------------------------+      +------------------------------------+
+    | Centralized Feature Flagging |      |               Strimzi              |
+    |            Server            |      | (Cluster, User and Topic Operator) |
+    +------------------------------+      +------------------------------------+
+                   |                                        |
+                   |                                        |
+                   |   <----------- API Calls ----------->  |
+
+Where in each component (i.e., ClusterOperator, UserOperator and TopicOperator), we will fetch feature flags dynamically from the OpenFeature API, which is managed by feature flagging server.
```

Comment:
What happens in the communication to the server fails? I guess we're unable to proceed with the reconciliation. Thus the availability of the Strimzi operators becomes coupled to the availability of this feature flagging server. I don't think that's a problem for the proposal, but calling out explicitly for users.

Reply:
> What happens in the communication to the server fails? I guess we're unable to proceed with the reconciliation. Thus the availability of the Strimzi operators becomes coupled to the availability of this feature flagging server

Well this can be handled I think because we would end up in a scenario where:
```java
2024-09-27 09:10:18 ERROR GrpcConnector:149 - failed to connect to event stream, exhausted retries
```
client (strimzi) wants to reconnect because he is getting a `GrpcConnector` error. We can catch that and use just the default value of `FEATURE_GATE`; therefore, we no longer depend on feature flagging server availability. And if such error would disappear (feature flagging system is up) we continue with a normal state.